lib/filewatch/watch.rb

```diff
@@ -117,127 +117,147 @@ def inode(path, stat)
     def each(&block)
       synchronized do
         return if @files.empty?
-
-        file_deleteable = []
-        # creates this array just once
-        watched_files = @files.values
-
-        # look at the closed to see if its changed
-        watched_files.select {|wf| wf.closed? }.each do |watched_file|
-          path = watched_file.path
-          begin
-            stat = watched_file.restat
-            if watched_file.size_changed? || watched_file.inode_changed?(inode(path,stat))
-              # if the closed file changed, move it to the watched state
-              # not to active state because we want to use MAX_OPEN_FILES throttling.
-              watched_file.watch
+        begin
+          file_deletable = []
```

Comment:
yeah - its a weird name - in my defence I did not want ppl to think I was deleting actual files. However `@files` is kinda wrong too.  I am inclined to leave it because in the new code `@files` is called `@store` and it points to a proper Store class that has an api for upsert, delete and enumerate.